app/services/pieces_justificatives_service.rb

```diff
@@ -107,21 +107,43 @@ def record_type
     end
   end
 
-  def self.generate_dossier_export(dossier)
-    pdf = ApplicationController
-      .render(template: 'dossiers/show', formats: [:pdf],
-              assigns: {
-                include_infos_administration: true,
-                dossier: dossier
-              })
-
-    FakeAttachment.new(
-      file: StringIO.new(pdf),
-      filename: "export-#{dossier.id}.pdf",
-      name: 'pdf_export_for_instructeur',
-      id: dossier.id,
-      created_at: dossier.updated_at
-    )
+  def self.generate_dossier_export(dossiers)
+    return [] if dossiers.empty?
+
+    pdfs = []
+
+    procedure = dossiers.first.procedure
+    tdc_by_id = TypeDeChamp
+      .where(revision_id: procedure.revisions.ids)
```

Comment:
```suggestion
      .joins(:revisions)
      .where(revisions: { id: procedure.revisions })
```

`type_de_champs.revision_id` n'est pas fiable et sera supprimé bientôt.

Reply:
bien vu, merci